src/app/pages/wallets/wallet-details/transaction-editor/pages-wallet-transaction-editor.component.ts

```diff
@@ -0,0 +1,90 @@
+import { Component, Inject } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { TServerEntityId } from 'src/app/common/http/common.http.types';
+import { checkInputError } from 'src/app/common/utils/forms/formUtils';
+import { WalletTransactionType } from 'src/app/domains/transactions/domains.transactions.constants';
+import { WalletsDetailsTransaction } from '../pages-wallet-details-item.model';
+import { IWalletTransactionModalFormType } from './pages-wallet-transaction.editor.types';
+
+export const regexAmount = /(^(\d{1,})$)|((^\d{1,})\.\d{1,2}$)/;
+
+@Component({
+  selector: 'app-transaction-editor',
+  templateUrl: './pages-wallet-transaction-editor.component.html',
+  styleUrls: [ './pages-wallet-transaction-editor.component.scss' ],
+})
+export class PagesWalletTransactionEditorComponent {
+  public selectTransactionsTypes: Record<string, WalletTransactionType> = {
+    'Income': WalletTransactionType.Incomes,
+    'Expense': WalletTransactionType.Expenses,
+  };
+
+  public form!: FormGroup<IWalletTransactionModalFormType>;
+
+  private transactionId: TServerEntityId | null = null;
+
+  public constructor(
+    private readonly dialogRef: MatDialogRef<PagesWalletTransactionEditorComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: WalletsDetailsTransaction,
+  ) {
+    this.transactionId = this.data.id;
+
+    this.form = new FormGroup<IWalletTransactionModalFormType>({
+      amount: new FormControl(this.data.amount === 0 ? null : this.data.amount, {
+        validators: [
+          Validators.required,
+          Validators.pattern(regexAmount),
+        ],
+        nonNullable: false,
+      }),
+      description: new FormControl(this.data.description),
+      date: new FormControl(this.data.date, {
+        validators: [
+          Validators.required,
+        ],
+        nonNullable: true,
```

Comment:
It is nullable, after removing value it should be null. Or for new transaction it also should be null.